### DISCUSSION TITLE
often stuck on loading

### DISCUSSION THREAD

**Discussion by kngharv** [libre-tube/LibreTube#1190] [Q&A] [answered]
Why LibreTube stuck on loading most of the time?  I am using the latest ver 0.5

The reason why I am asking this is 
1. apparently, I am the only person who is running into this problem, judging the fact that no one complained in the discussion and/or no bug report filed against it
2. it worked every once a while, even though I have not done anything differently.

Going through some document, it seems that this LibreTube rely on Piped API.  I have no idea what they are, but TeamPipe offer a self-hosted option. 

Does it mean that the reason why I can't get LibreTube to work most of the time is due to the fact that the public instance of Piped "server" is simply too busy?

Thanks in advance

**Comment by Bnyro** [comment, marked as answer]
Yes, that's probably due to the workload of the instance you are using and its location.
LibreTube doesn't connect to YouTube servers directly, there's always a Piped instance in-between, therefore your IP as well as other personal data can't be accessed by YouTube and you're experience is more private. You can find more about Piped and its infrastructure [here](https://piped-docs.kavin.rocks/docs/architecture/) and in [Piped's README](https://github.com/TeamPiped/Piped).

You can change the server in the "instance" settings of the app, a list of public instances can be found [here](https://piped.kavin.rocks/preferences).

The official one has the largest amount of nodes but the most users at the same time, you might find one that is faster for you while being less utilized.

Regards

**Reply by kngharv** [reply]
Thank you.

I will see if I can host my own piped node later.  :)

**Reply by Bnyro** [reply]
Feel free to ask in the [Piped Matrix room](https://matrix.to/#/#piped:matrix.org) if you have any questions :)

**Reply by alefvanoon** [reply]
> Thank you.
> 
> I will see if I can host my own piped node later. :)

https://piped-docs.kavin.rocks/docs/self-hosting/
